components/kyma-environment-broker/internal/reconciler/fake.go

```diff
@@ -128,11 +128,11 @@ func (c *FakeClient) createOrUpdate(cluster Cluster) (*State, error) {
 	// cluster exists in db - add new configuration version
 	//TODO: implement comparision mechanism for configs (new config should not be added if nothing changes in request) - needed for upgrade testing
 	//TODO: implement clusterVersion bumping (happens when Kyma version is updated?)
-	latestConfigVersion := int64(len(c.inventoryClusters[cluster.Cluster].clusterStates))
+	latestConfigVersion := int64(len(c.inventoryClusters[cluster.Cluster].clusterStates)) + 1
```

Comment:
```suggestion
	latestConfigVersion := int64(len(c.inventoryClusters[cluster.Cluster].clusterStates)) + 1
	
	c.inventoryClusters[cluster.Cluster].clusterConfigs[latestConfigVersion] = cluster
```

adding this line would allow us to keep track of configurations passed to fake-reconciler with `ApplyClusterConfiguration` calls for easier debugging

Reply:
thanks, I have rebased on top of https://github.com/kyma-project/control-plane/pull/1166, I think that also includes this change